### PR TITLE
Enhance IborCapFloorTrade pricer

### PR DESCRIPTION
**Pull request by skitini**
- Add implied volatility and forward rate calculations to IBOR cap/floor trade, product, leg and period pricers.
- For the trade, product and leg pricers, a map is returned of the implied volatility / forward rate per individual caplet/floorlet.

### CODE REVIEW

**Review by benhamilton-brown — COMMENTED** [review]
All looks fine, is it worth adding a few tests for some floor/floorlets?

**Review by benhamilton-brown — APPROVED** [review]

File: modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/VolatilityIborCapFloorLegPricer.java [hunk 1]
```diff
@@ -227,6 +230,44 @@ public CurrencyAmount currentCash(
...
 
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the forward rates for each caplet/floorlet of the Ibor cap/floor leg.
+   *
+   * @param capFloorLeg  the Ibor cap/floor leg
+   * @param ratesProvider  the rates provider
+   * @return the forward rates
+   */
+  public Map<IborCapletFloorletPeriod, Double> forwardRates(
```
Comment (jodastephen) [inline modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/VolatilityIborCapFloorLegPricer.java:241]: Maybe there should be a bean to represent this map?

Comment (skitini) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/VolatilityIborCapFloorLegPricer.java:241]: Would you recommend two beans, `IborCapletFloorletPeriodForwardRates` and `IborCapletFloorletPeriodVolatilities`, or just one bean that handles both, as they both just a map of period to double?

Comment (jodastephen) [reply modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/VolatilityIborCapFloorLegPricer.java:241]: We tend to share classes in results, eg `CurrencyAmount`. So I think `IborCapletFloorletPeriodAmounts`

**Review by jodastephen — APPROVED** [review]
I suspect it may be possible to add methods to the new class, eg. ones that return `double` instead of `Double`, but experience can guide us with this.

**Pull request merged**